BAGIAN 4 (A)

In [9]:
# Import library yang diperlukan
import pandas as pd
from scipy.stats import ttest_ind

# Load data Titanic
file_path = 'train.csv'
data = pd.read_csv(file_path)

# Menampilkan beberapa baris data untuk memastikan formatnya
data.head()

# Menghapus data yang memiliki nilai Age yang hilang
data = data.dropna(subset=['Age'])

# Memisahkan data berdasarkan status kelangsungan hidup
survived = data[data['Survived'] == 1]['Age']
not_survived = data[data['Survived'] == 0]['Age']

# Melakukan uji t independen
t_stat, p_value = ttest_ind(survived, not_survived, equal_var=False)

# Menampilkan hasil uji t
print("Hasil Uji T:")
print(f"T-Statistic: {t_stat:.4f}")
print(f"P-Value: {p_value:.4f}")

Hasil Uji T:
T-Statistic: -2.0460
P-Value: 0.0412


1. Hasil Uji T:
    - T-Statistic: -2.0460
    - P-Value: 0.0412

2. Hipotesis:
    - H0 (Null Hypothesis): Tidak terdapat perbedaan usia yang signifikan antara penumpang yang selamat dan tidak selamat.
    - H1 (Alternative Hypothesis): Terdapat perbedaan usia yang signifikan antara penumpang yang selamat dan tidak selamat.

3. Keputusan:
    - Dengan tingkat signifikansi (𝛼) = 0.05:
    - P-value = 0.0412 < 0.05

4. Kesimpulan:
Terdapat perbedaan usia yang signifikan secara statistik antara penumpang yang selamat dan tidak selamat pada kapal Titanic.
menolak Null Hypothesis karena ada bukti statistik bahwa usia penumpang yang selamat berbeda secara signifikan dari usia penumpang yang tidak selamat.

BAGIAN 4 (B)

In [10]:
from scipy.stats import chi2_contingency

# Membuat tabel kontingensi berdasarkan jenis kelamin dan kelangsungan hidup
contingency_table = pd.crosstab(data['Sex'], data['Survived'])

# Menampilkan tabel kontingensi
print("Tabel Kontingensi:")
print(contingency_table)

# Melakukan uji chi-square
chi2_stat, p_value, dof, expected = chi2_contingency(contingency_table)

# Menampilkan hasil uji chi-square
print("\nHasil Uji Chi-Square:")
print(f"Chi-Square Statistic: {chi2_stat:.4f}")
print(f"P-Value: {p_value:.4e}")
print(f"Degrees of Freedom: {dof}")
print("\nExpected Frequencies:")
print(expected)

Tabel Kontingensi:
Survived    0    1
Sex               
female     64  197
male      360   93

Hasil Uji Chi-Square:
Chi-Square Statistic: 205.0258
P-Value: 1.6717e-46
Degrees of Freedom: 1

Expected Frequencies:
[[154.99159664 106.00840336]
 [269.00840336 183.99159664]]


1. HASIL KEPUTUSAN
- menolak null hipotesis (H0), yang menyatakan bahwa tidak terdapat hubungan signifikan antara jenis kelamin dan kelangsungan hidup. Sebaliknya, data mendukung hipotesis alternatif (H1), yaitu terdapat hubungan yang signifikan antara jenis kelamin dan kelangsungan hidup.

BAGIAN 4 (C)

In [11]:
# Import library yang diperlukan
from scipy.stats import chi2_contingency, ttest_ind

# Uji Chi-Square untuk jenis kelamin dan kelangsungan hidup
contingency_table_sex = pd.crosstab(data['Sex'], data['Survived'])
chi2_stat_sex, p_value_sex, dof_sex, expected_sex = chi2_contingency(contingency_table_sex)

# Menampilkan hasil uji Chi-Square
print("Hasil Uji Chi-Square (Jenis Kelamin):")
print(f"Chi-Square Statistic: {chi2_stat_sex:.4f}")
print(f"P-Value: {p_value_sex:.4e}")
print(f"Degrees of Freedom: {dof_sex}")
print("\nExpected Frequencies:")
print(expected_sex)

# Uji T untuk usia dan kelangsungan hidup
data_age = data.dropna(subset=['Age'])  # Menghapus data dengan usia yang hilang
survived_age = data_age[data_age['Survived'] == 1]['Age']
not_survived_age = data_age[data_age['Survived'] == 0]['Age']
t_stat_age, p_value_age = ttest_ind(survived_age, not_survived_age, equal_var=False)

# Menampilkan hasil uji T
print("\nHasil Uji T (Usia):")
print(f"T-Statistic: {t_stat_age:.4f}")
print(f"P-Value: {p_value_age:.4f}")

Hasil Uji Chi-Square (Jenis Kelamin):
Chi-Square Statistic: 205.0258
P-Value: 1.6717e-46
Degrees of Freedom: 1

Expected Frequencies:
[[154.99159664 106.00840336]
 [269.00840336 183.99159664]]

Hasil Uji T (Usia):
T-Statistic: -2.0460
P-Value: 0.0412


1. HASIL KEPUTUSAN
- Karena kedua analisis (chi-square untuk jenis kelamin dan uji t untuk usia) menunjukkan hasil yang signifikan (𝑃<0.05P<0.05), maka menolak null hipotesis (H0).